src/improvements/tasks/StateOverrideManager.sol

```diff
@@ -23,26 +21,30 @@ abstract contract StateOverrideManager {
     Simulation.StateOverride[] private _stateOverrides;
 
     /// @notice Get all state overrides for simulation. Combines default Tenderly overrides
-    /// with user-defined overrides.
-    function getStateOverrides(address parentMultisig, uint256 parentMultisigNonce)
-        public
-        view
-        returns (Simulation.StateOverride[] memory)
-    {
-        // Create default Tenderly override (sets nonce, threshold, and makes msg.sender an owner)
-        Simulation.StateOverride memory defaultOverride =
-            _createDefaultTenderlyOverride(parentMultisig, parentMultisigNonce);
-
-        // Combine default override with user-defined overrides
-        Simulation.StateOverride[] memory allOverrides = new Simulation.StateOverride[](1 + _stateOverrides.length);
-        allOverrides[0] = defaultOverride;
-
-        // Add user-defined overrides (these take precedence over default ones)
-        for (uint256 i = 0; i < _stateOverrides.length; i++) {
-            allOverrides[i + 1] = _stateOverrides[i];
+    /// with user-defined overrides. User defined overrides are applied last.
+    /// If a child multisig is provided then we are working with a nested safe.
+    /// In this case we need additional state overrides.
+    function getStateOverrides(
+        address parentMultisig,
+        uint256 parentMultisigNonce,
+        address optionalChildMultisig,
+        uint256 optionalChildMultisigNonce
+    ) public view returns (Simulation.StateOverride[] memory allOverrides_) {
+        if (optionalChildMultisig != address(0)) {
+            allOverrides_ = new Simulation.StateOverride[](2 + _stateOverrides.length);
+            allOverrides_[0] = _parentMultisigTenderlyOverride(parentMultisig, parentMultisigNonce);
+            allOverrides_[1] = _childMultisigTenderlyOverride(optionalChildMultisig, optionalChildMultisigNonce);
+            // Add user-defined overrides (these take precedence over default ones)
+            for (uint256 i = 0; i < _stateOverrides.length; i++) {
+                allOverrides_[i + 2] = _stateOverrides[i];
+            }
```

Comment:
In forge adding user-defined overrides last means they take precedence because they get `vm.store`d last, but does Tenderly behave the same way if you have conflicting overrides?

Reply:
Yes, I recall observing this behavior and I noticed that in base/contract they also follow this precedence approach by putting the simulation overrides last: https://github.com/base/contracts/blob/6ca464d118d2d33331ea44893f0930f753663f60/script/universal/NestedMultisigBase.sol#L124-L133